tfhe/src/integer/server_key/radix_parallel/tests_cases_unsigned.rs

```diff
@@ -0,0 +1,2825 @@
+            assert_eq!(ct_res, tmp);
+            clear = (clear + clear_1) % modulus;
+
+            let dec_res: u64 = cks.decrypt(&ct_res);
+            assert_eq!(clear, dec_res);
+        }
+    }
+}
+
+pub(crate) fn default_scalar_sub_test<P, T>(param: P, mut executor: T)
+where
+    P: Into<PBSParameters>,
+    T: for<'a> FunctionExecutor<(&'a RadixCiphertext, u64), RadixCiphertext>,
+{
+    let (cks, mut sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    sks.set_deterministic_pbs_execution(true);
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+
+    let mut rng = rand::thread_rng();
+
+    executor.setup(&cks, sks.clone());
+
+    let mut clear;
+
+    for _ in 0..NB_TEST_SMALLER {
+        let clear_0 = rng.gen::<u64>() % modulus;
+        let clear_1 = rng.gen::<u64>() % modulus;
+
+        let ctxt_0 = cks.encrypt(clear_0);
+
+        let mut ct_res = executor.execute((&ctxt_0, clear_1));
+        assert!(ct_res.block_carries_are_empty());
+
+        clear = (clear_0.wrapping_sub(clear_1)) % modulus;
+
+        // Sub multiple times to raise the degree
+        for _ in 0..NB_TEST_SMALLER {
+            let tmp = executor.execute((&ct_res, clear_1));
+            ct_res = executor.execute((&ct_res, clear_1));
+            assert!(ct_res.block_carries_are_empty());
+            assert_eq!(ct_res, tmp);
+            clear = (clear.wrapping_sub(clear_1)) % modulus;
+
+            let dec_res: u64 = cks.decrypt(&ct_res);
+            assert_eq!(clear, dec_res);
+        }
+    }
+}
+
+pub(crate) fn default_small_scalar_mul_test<P, T>(param: P, mut executor: T)
+where
+    P: Into<PBSParameters>,
+    T: for<'a> FunctionExecutor<(&'a RadixCiphertext, u64), RadixCiphertext>,
+{
+    let (cks, mut sks) = KEY_CACHE.get_from_params(param);
+    let cks = RadixClientKey::from((cks, NB_CTXT));
+
+    sks.set_deterministic_pbs_execution(true);
+
+    let mut rng = rand::thread_rng();
+
+    // message_modulus^vec_length
+    let modulus = cks.parameters().message_modulus().0.pow(NB_CTXT as u32) as u64;
+
+    let scalar_modulus = cks.parameters().message_modulus().0 as u64;
+
+    executor.setup(&cks, sks);
+
+    let mut clear_res;
+    for _ in 0..NB_TEST_SMALLER {
+        let clear = rng.gen::<u64>() % modulus;
+        let scalar = rng.gen::<u64>() % scalar_modulus;
+
+        let ct = cks.encrypt(clear);
+
+        let mut ct_res = executor.execute((&ct, scalar));
+        assert!(ct_res.block_carries_are_empty());
+
+        clear_res = clear * scalar;
+        for _ in 0..NB_TEST_SMALLER {
+            // scalar multiplication
+            let tmp = executor.execute((&ct_res, scalar));
+            ct_res = executor.execute((&ct_res, scalar));
+            assert!(ct_res.block_carries_are_empty());
+            assert_eq!(tmp, ct_res);
+            clear_res *= scalar;
```

Comment:
same thing potentially wrapping (I may have missed that in other parts of this PR)

not a blocker can be done in a follow up issue